### PR TITLE
Close IMAP connection in case of parse error

### PR DESCRIPTION
**Pull request by disaykin**

### CODE REVIEW

**Comment by dinhviethoa** [comment]
Could you take a look at how it's done for socket errors?
https://github.com/MailCore/mailcore2/blob/master/src/core/imap/MCIMAPSession.cpp#L1652
We could rather do something similar for parse errors.

**Comment by disaykin** [comment]
Done

File: src/async/imap/MCIMAPOperation.cpp [hunk 1]
```diff
@@ -110,14 +110,18 @@ void IMAPOperation::start()
         IMAPAsyncConnection * connection = mMainSession->sessionForFolder(mFolder, mUrgent);
         setSession(connection);
     }
-    mSession->runOperation(this);
+    session()->runOperation(this);
 }
 
+namespace {
```
Comment (dinhviethoa) [inline src/async/imap/MCIMAPOperation.cpp]: I know that is a good best practice in C++, though could you remove anyway that anonymous namespace anyway?

**Comment by dinhviethoa** [comment]
One small change. Everything else looks good!

**Comment by disaykin** [comment]
Done.

**Comment by dinhviethoa** [comment]
Thanks!

**Pull request merged**